game.py

```diff
@@ -0,0 +1,302 @@
+
+    else:
+        for move in game_state.get_next_moves(moves_sort = sort):
+            ## print ('min board prior to move  ', game_state.get_board())
+            new_game_state = game_state.make_move(move)
+            # print ('min board after move  ', new_game_state.get_board())
+            move_and_score = minimax(new_game_state, look_forward - 1, player_next_first, alpha, beta, sort_depth)  
+            # print ('Min score : ', move_and_score[1], 'beta = ', beta)            
+            if move_and_score[1] < beta:
+                beta = move_and_score[1]
+                best_move = move
+                # print('Best move min : ', move)
+            print('min alpha = ', alpha, 'min beta = ', beta)
+            if beta <= alpha:
+                print ('Break min')
+                break
+            
+        return best_move, beta
+    
+def get_computer_move(game_state):
+    player_next_first = game_state.get_player_next()
+    look_forward = choose_look_ahead()
+    sort_depth = get_sort_depth()
+    score_count = 0
+    best_move_and_score = minimax(game_state, look_forward, player_next_first, -inf, inf, sort_depth)
+    return best_move_and_score
+
+
+def choose_board_size():
+    rows = int(input("Enter number of rows : "))
+    columns = int(input("Enter number of columns : "))
+    return (rows, columns)
+
+def create_board(board_size):
+    board = [[None] * int(board_size[1]) for _ in range(int(board_size[0]))]
+    return board
+
+def choose_look_ahead():
+    look_ahead = int(input("Enter look ahead : "))
+    return look_ahead
+
+def move_more():
+    more_moves = input("Another Move? (Y/N) : ")
+    if more_moves == 'Y' or more_moves == 'y':
+        return True
+    else:
+        return False
+
+def undo_move():
+    undo_last_move = input("Undo last move? (Y/N) : ")
+    if undo_last_move == 'Y' or undo_last_move == 'y':
+        return True
+    else:
+        return False
+
+def get_sort_depth():
+    sort_depth = int(input("Enter sort depth : "))
+    return sort_depth
+
+def human_to_play():
+    human = input("Human to play? (Y/N) : ")
+    if human == 'Y' or human == 'y':
+        return True
+    else:
+        return False
+
+def get_human_move(game_state):
+    valid = False
+    while not valid:
+        row = int(input("Enter row : ")) - 1
+        column = int(input("Enter column : ")) - 1
+        if  game_state.valid_move((row, column)):
+            valid = True
+        else:
+            print('Invalid move. Try again')
+            valid = False
+    return row, column
+
+def get_hint():
+    hint = input("Get hint? (Y/N) : ")
+    if hint == 'Y' or hint == 'y':
+        return True
+    else:
+        return False
+
+
+def main():
+    board = []
+    board_size = choose_board_size()
+    print('board size : ', board_size)
+    board = create_board(board_size)
+    print('board : ', board)
+    
+    player_next = 1
+    number_of_moves = 0
+    
+    global score_count

```

Comment:
I think using a global here was great! There are ways to hook up this information in a cleaner way, but almost certainly not worth the effort at this point. The way I'd do this if I still wanted access to thecount and the application was getting large enough that the global-variable was too cumbersome, I'd use a closure-wrapper for the score-function and then some method-binding to add it to an existing object of GameState.

Might be worth revisiting at some point, since the wrapping and binding stuff showcases the flexibility of Python that sets it apart from other more rigid languages.

This is only a rough example but along the line of thinking I'm referring to:
```
def say_hello():
  print("hello")

def wrap_count_calls(fn):
  count = 0
  def wrapped_fn():
    wrapped_fn.count += 1
    fn()
  wrapped_fn.count = 0
  return wrapped_fn

say_hello_special = wrap_count_calls(say_hello)

for _ in range(10):
  say_hello_special()

print(f"I just printed hello {say_hello_special.count} times!")
```